flybywiresim/a32nx
# fix(ecam): Fix HYD PTU message on ECAM

Fixes #5077 

## Summary of Changes
The logic for when "HYD PTU" showed on the ECAM was incorrect. More correctly, the memo should now show whenever the PTU is on according to the new hydraulics system implementation.

This is only a temporary solution until the FWC/FWS is implemented properly. This is not an optimal solution. If the performance hit is too hard. Feel free to just reject the PR, so the logic can be implemented properly later on.

## Screenshots (if necessary)
HYD PTU message is no longer shown after engine 1 start:
![eng-start-no-ptu](https://user-images.githubusercontent.com/22713769/121770415-7d5a3300-cb69-11eb-9060-85f0db9406e0.png)
HYD PTU message is shown with 2 engines, but yellow hydraulics pump turned off and PTU on HYD ECAM page (also works for other side):
![yellow-hyd-pump-off-ptu](https://user-images.githubusercontent.com/22713769/121770478-c5795580-cb69-11eb-80e6-beee05da18cd.png)

## References

- @raeliass who filed the issue is a type rated A320 pilot linked references in issue #5077.




## Additional context



Discord username (if different from GitHub): BlueberryKing#6641

## Testing instructions
- Spawn in C/D, go through normal preflight.
- Perform a single engine taxi out procedure:
  - Start engine 1
  - Set ignition normal
  - Set APU bleed off
  - Set Yellow Hyd eletric pump ON
- At no point during this should the HYD PTU memo show on the ECAM, neither should the PTU come active on the lower ECAM HYD page.
- Turn off yellow electric pump
- Start engine 2
- Turn off engine 2 hydraulics pump (yellow) on the overhead panel, verify PTU comes on on lower ECAM and HYD PTU message shows.
- Explicitly turn off PTU valve on the overhead panel, verify PTU turns off on lower ECAM and HYD PTU message is not shown.
- Turn PTU valve back on
- Turn on yellow electric pump on the overhead panel, verify PTU turns off on lower ECAM and HYD PTU message is not shown. 
- Turn off yellow electric pump
- Turn off engine 1 hydraulics pump (green) on the overhead panel, verify PTU comes on on lower ECAM and HYD PTU message shows.

I am not very familiar with the hydraulics system, so I would be thankful if anyone testing the PR could also try other known scenarios where they would expect the PTU to come on/stay off.


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
